src/porepy/models/abstract_equations.py

```diff
@@ -183,3 +184,57 @@ def perturbation_from_reference(self, variable_name: str, grids: list[pp.Grid]):
         d_var = var(grids) - var_ref(grids)
         d_var.set_name(variable_name + "_perturbation")
         return d_var
+
+
+class BoundaryConditionsMixin:
+    """TODO"""
+
+    mdg: pp.MixedDimensionalGrid
+
+    def update_boundary_conditions(self, initial: bool) -> None:
+        """This method is called before a new time step to set the values of the
+        boundary conditions. The specific boundary condition values should be updated in
+        concrete overrides. By default, it does nothing.
+
+        Note:
+            One can use the convenience method `_update_boundary_condition` for each
+            boundary condition value.
+        """
+
+    def _update_boundary_condition(
+        self,
+        name: str,
+        function: Callable[[Sequence[pp.BoundaryGrid]], np.ndarray],
+        initial: bool,
+    ) -> None:
+        """TODO
+
+        TODO: Somewhere we must have an assertion that there is only 1 time step behind
+        stored, and 1 iterate.
+        """
+
+        for bg, data in self.mdg.boundaries(return_data=True):
+            # Set the known time step value.
+            if initial:
+                vals = function([bg])
+                pp.set_solution_values(
+                    name=name, values=vals, data=data, time_step_index=0
+                )
+            else:
+                vals = data[pp.ITERATE_SOLUTIONS][name][0]
```

Comment:
@IngridKJ, should we make a get_solution_values method?

Reply:
Sounds reasonable. I assume it should/can be analogous to the pp.set_solution_values()? If so, I already have one in my private repo. Should be super quick to add to porepy

Reply:
That would be nice. If you do, please check for occurrences  of pp.ITERATE_SOLUTIONS and pp.TIME_STEP_SOLUTIONS throughout the code and replace by new function where appropriate.

Reply:
Sure. I take it the method is to be part of this PR? And are there any suggestions about where to put it?

Reply:
No, separate PR. Should go same place at the set method, shouldn't it?

Reply:
I agree they should be together, question is rather if we want them both within equation_system.py or not. equation_system.py is where the set function is now. But I will start working on an issue/PR for this, and we can discuss placement of the functions further elsewhere (if there is anything to discuss)